zkevm-circuits/src/witness/block.rs

```diff
@@ -271,6 +274,51 @@ impl Block {
         log::debug!("tx_log num: {}", self.rws.rw_num(RwTableTag::TxLog));
         log::debug!("start num: {}", self.rws.rw_num(RwTableTag::Start));
     }
+
+    // This helper returns bytecodes's whether `code_hash` is belong to first bytecode circuit.
+    #[cfg(feature = "dual_bytecode")]
+    pub(crate) fn is_first_bytecode(&self, code_hash: &U256) -> bool {
```

Comment:
we can add a witness/bytecode.rs and move these functions there?

Reply:
there is already bytecode file `witness/bytecode.rs` which handle single `Bytecode` struct, like `table_assignments` , `make_push_rlc` etc.  dual bytecode sub circuit methods (like `is_first_sub_bytecode_circuit`, `get_bytecodes_for_dual_sub_circuits` are related to block's `bytecode_map` info,  not single `Bytecode` context,  more close to block context,  feeling keep them here not bad.